p2p/libp2p/metrics/factory/connectionWatcherFactory.go

```diff
@@ -0,0 +1,24 @@
+package factory
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/p2p/libp2p/metrics"
+)
+
+const printConnectionsWatcher = "print"
+const disabledConnectionsWatcher = "disabled"
+
+// NewConnectionsWatcher creates a new ConnectionWatcher instance based on the input parameters
+func NewConnectionsWatcher(connectionsWatcherType string, timeToLive time.Duration) (p2p.ConnectionsWatcher, error) {
+	switch connectionsWatcherType {
+	case printConnectionsWatcher:
+		return metrics.NewPrintConnectionsWatcher(timeToLive)
+	case disabledConnectionsWatcher:
+		return metrics.NewDisabledConnectionsWatcher(), nil
+	default:
```

Comment:
I would return the disabled connections watcher if the config's value isn't right

Reply:
might hide improper configuration leading to missing or badly configured p2p.toml file